skyportal/handlers/api/unsourced_finder.py

```diff
@@ -0,0 +1,301 @@
+        - in: query
+          name: type
+          nullable: true
+          schema:
+            type: string
+            enum: [png, pdf]
+          description: |
+            output type
+        - in: query
+          name: num_offset_stars
+          schema:
+            type: integer
+            minimum: 0
+            maximum: 4
+          description: |
+            output desired number of offset stars [0,4] (default: 3)
+        responses:
+          200:
+            description: A PDF/PNG finding chart file
+            content:
+              application/pdf:
+                schema:
+                  type: string
+                  format: binary
+              image/png:
+                schema:
+                  type: string
+                  format: binary
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+        location_type = self.get_query_argument('location_type')
+        if location_type not in ["gaia_edr3", "gaia_dr2", "pos"]:
+            return self.error(f'Invalid argument for `location_type`: {location_type}')
+
+        obstime = self.get_query_argument(
+            'obstime', datetime.datetime.utcnow().isoformat()
+        )
+        if not isinstance(isoparse(obstime), datetime.datetime):
+            return self.error('obstime is not valid isoformat')
+
+        catalog_id = self.get_query_argument('catalog_id', 'unknown')
+        if location_type != "pos":
+            # database name should be something like gaiaedr3
+            db_name = "".join(location_type.split("_"))
+            obstime_decimalyear = Time(isoparse(obstime)).decimalyear
+            query_string = f"""
+                SELECT source_id, ra, dec, pmra, pmdec, coord1(prop) AS ra_obs, coord2(prop) AS dec_obs FROM (
+                    SELECT gaia.source_id, ra, dec, pmra, pmdec,
+                    EPOCH_PROP_POS(ra, dec, parallax, pmra, pmdec, 0, ref_epoch, {obstime_decimalyear}) AS prop
+                    FROM {db_name}.gaia_source AS gaia
+                WHERE gaia.source_id={catalog_id}
+                ) AS subquery
+            """
+            g = GaiaQuery()
+            r = g.query(query_string)
+            if len(r) != 1:
+                return self.error(
+                    f'Cannot get position information for `catalog_id` = {catalog_id}'
+                )
+            ra = r["ra_obs"].data[0]
+            dec = r["dec_obs"].data[0]
+            obj_id = f'{location_type.split("_")[-1]} {catalog_id}'
+            pmra, pmdec = ma.getdata(r["pmra"])[0], ma.getdata(r["pmdec"])[0]
+            extra_display_string = f'{pmra:0.4} E \u2033/yr {pmdec:0.4} N \u2033/yr'
+        else:
+            ra = self.get_query_argument('ra')
+            try:
+                ra = float(ra)
+            except ValueError:
+                # could not handle inputs
+                return self.error('Invalid argument for `ra`')
```

Comment:
Also add a check for the range:
```suggestion
                return self.error('Invalid argument for `ra`')
            if not 0 <= ra < 360.0:
                return self.error("Invalid value for `ra`: must be 0 <= ra < 360.0")
```